aws-s3-transfer-manager/src/operation/download/service.rs

```diff
@@ -139,12 +148,25 @@ pub(super) fn distribute_work(
 
         let svc = svc.clone();
         let comp_tx = comp_tx.clone();
+        let cancel_tx = ctx.state.cancel_tx.clone();
 
         let task = async move {
-            // TODO: If downloading a chunk fails, do we want to abort the download?
             let resp = svc.oneshot(req).await;
+            if let Err(err) = &resp {
+                if *err.kind() != ErrorKind::OperationCancelled && cancel_tx.send(true).is_err() {
+                    tracing::debug!(
+                        "all receiver ends have dropped, unable to send a cancellation signal"
+                    );
+                }
+            }
+
             if let Err(err) = comp_tx.send(resp).await {
```

Comment:
Is it OK that ErrorKind::OperationCancelled` gets sent on `comp_tx`? That could result in MANY chunks sending their cancellation error on the chunk transmitter

Do all of these end up reported to the user via the Body/Output iterator? Or does it filter errors so the user only sees the 1st? Can we guarantee that the ACTUAL error gets reported to the user before any cancellation errors?

Reply:
As soon as we encounter an error, we close the channel so user will only see the first error. 

> Can we guarantee that the ACTUAL error gets reported to the user before any cancellation errors?

Tokio does guarantee that "All data sent on Sender will become available on Receiver in the same order as it was sent." However, if the channel is full, all of them will be waiting at an await point. In that case (not 100% sure), the order won't be guaranteed. We will need a fair bit of complexity to work around that, i.e., maybe collect all the errors and then only send those that are not `OperationCancelled` errors. Not sure if there is a simpler way to do this. @aajtodd any ideas?

Reply:
Do we need to send operation cancelled across the channel at all?

Reply:
I was thinking there can be a case where only the operation cancelled error is there, but yeah, we can just not send that error at all.

Reply:
A user wouldn't act on that error right? I wouldn't think so anyway, it doesn't tell them anything they don't already know if they cancelled the operation, thus it probably doesn't make sense to yield it to them.

Reply:
Thanks, I have updated it to ignore `OperationCancelled` errors.